src/main/java/com/google/sps/model/ActionableMessageHelperImpl.java

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.model;
+
+import com.google.api.services.gmail.model.Message;
+import com.google.api.services.gmail.model.MessagePartHeader;
+import com.google.sps.exceptions.GmailMessageFormatException;
+import com.google.sps.utility.GmailUtility;
+import java.util.Arrays;
+import java.util.List;
+
+/** Contains methods that help create the non-trivial fields in an ActionableMessage. */
+public final class ActionableMessageHelperImpl implements ActionableMessageHelper {
+  // Public for testing purposes
+  public static final String UNDISCLOSED_RECIPIENTS_TO_HEADER = "undisclosed-recipients:;";
+  public static final int MAILING_LIST_LOWER_BOUND = 15;
+
+  @Override
+  public ActionableMessage.MessagePriority assignMessagePriority(
+      Message message, String userEmail) {
+    if (isStarred(message)) {
+      return ActionableMessage.MessagePriority.HIGH;
+    }
+
+    if (!isFromMailingList(message, userEmail) || isImportant(message)) {
+      return ActionableMessage.MessagePriority.MEDIUM;
+    }
+
+    return ActionableMessage.MessagePriority.LOW;
+  }
+
+  /**
+   * If "STARRED" is included in a message's labelIds, it has been starred by the user
+   *
+   * @param message Message object to be checked
+   * @return true if starred, false otherwise
+   */
+  private boolean isStarred(Message message) {
+    List<String> labelIds = message.getLabelIds();
+    return labelIds != null && labelIds.contains("STARRED");
+  }
+
+  /**
+   * If the user's email is not in the "To" header, or if there are more than 15 recipients, the
```

Comment:
nit: `MAILING_LIST_LOWER_BOUND` instead of `15`

Reply:
Agreed - thanks!